metadata.go

```diff
@@ -51,3 +63,58 @@ func (m *AppImageMetadata) metadataForBuildpack(id string) BuildpackMetadata {
 	}
 	return BuildpackMetadata{}
 }
+
+func (m *CacheImageMetadata) metadataForBuildpack(id string) BuildpackMetadata {
+	for _, bpMd := range m.Buildpacks {
+		if bpMd.ID == id {
+			return bpMd
+		}
+	}
+	return BuildpackMetadata{}
+}
+
+func getAppMetadata(image image.Image, log *log.Logger) (AppImageMetadata, error) {
+	metadata := AppImageMetadata{}
+	contents, err := getMetadata(image, MetadataLabel, log)
+	if err != nil {
+		return metadata, err
+	}
+
+	if err := json.Unmarshal([]byte(contents), &metadata); err != nil {
+		log.Printf("WARNING: image '%s' has incompatible '%s' label\n", image.Name(), MetadataLabel)
+		return metadata, nil
```

Comment:
Seems like we shouldn't be returning a metadata with undefined contents + no error here. Should this be an empty `AppImageMetadata` struct?